spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcValueReaders.java

```diff
@@ -170,6 +177,31 @@ public UTF8String nonNullRead(ColumnVector vector, int row) {
     }
   }
 
+  private static class UUIDReader implements OrcValueReader<UTF8String> {
+    private static final ThreadLocal<ByteBuffer> BUFFER =
+        ThreadLocal.withInitial(
+            () -> {
+              ByteBuffer buffer = ByteBuffer.allocate(16);
+              buffer.order(ByteOrder.BIG_ENDIAN);
+              return buffer;
+            });
+
+    private static final UUIDReader INSTANCE = new UUIDReader();
+
+    private UUIDReader() {}
+
+    @Override
+    public UTF8String nonNullRead(ColumnVector vector, int row) {
+      BytesColumnVector bytesVector = (BytesColumnVector) vector;
```

Comment:
Do we need to put this into a buffer? Just wondering if we should just use the array itself?

Reply:
I was mainly aligning with other places in the code that also used a Thread local ByteBuffer when reading UUIds, but we might be able to use the array itself (need to double-check that)

Reply:
Using the array directly would be ideal.

Reply:
done